### PR TITLE
Feature/4174 diary edit entries screen

### PR DESCRIPTION
**Pull request by nickguendling**
## Description
Adds a screen to manage all contact persons or locations.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4174

## Screenshots
Simulator Screen Shot - iPhone 12 Pro Max - 2020-12-15 at 15 51 06
Simulator Screen Shot - iPhone 12 Pro Max - 2020-12-15 at 15 51 09
Simulator Screen Shot - iPhone 12 Pro Max - 2020-12-15 at 15 51 24
Simulator Screen Shot - iPhone 12 Pro Max - 2020-12-15 at 15 51 30
Simulator Screen Shot - iPhone 12 Pro Max - 2020-12-15 at 15 51 51
Simulator Screen Shot - iPhone 12 Pro Max - 2020-12-15 at 15 52 00
Simulator Screen Shot - iPhone 12 Pro Max - 2020-12-15 at 15 52 14
Simulator Screen Shot - iPhone 12 Pro Max - 2020-12-15 at 15 52 30

### CODE REVIEW

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/EditEntries/DiaryEditEntriesViewController.swift [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+	// MARK: - Private
+
+	private let viewModel: DiaryEditEntriesViewModel
+	private let onCellSelection: (DiaryEntry) -> Void
+	private let onDismiss: () -> Void
+
+	private var subscriptions = [AnyCancellable]()
+
+	private var shouldReload = true
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/EditEntries/DiaryEditEntriesViewController.swift:115]: why not give the ViewModel a closure to refresh the tableView? So the ViewModel can decide when it wan't to reload the table view.
The only problem then will be: you can't init the ViewModel completely from outside the controller.

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/EditEntries/DiaryEditEntriesViewController.swift [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+	@IBOutlet weak var tableView: UITableView!
+	@IBOutlet weak var deleteAllButton: ENAButton!
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/EditEntries/DiaryEditEntriesViewController.swift:118]: private

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/EditEntries/DiaryEditEntriesViewController.swift [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+		tableView.dataSource = self
+
+		tableView.separatorStyle = .none
+		tableView.rowHeight = UITableView.automaticDimension
+		tableView.estimatedRowHeight = 60
+
+		tableView.isEditing = true
+	}
+
+	@IBAction func didTapDeleteAllButton(_ sender: ENAButton) {
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/EditEntries/DiaryEditEntriesViewController.swift:136]: private

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/EditEntries/DiaryEditEntriesViewController.swift (hunk 1 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/EditEntries/DiaryEditEntriesViewController.swift:115]: Well, we already have combine for this binding, that's why I haven't introduces a closure. The only problem is the animations that get cut off if I don't prevent the reload. I think that is a problem of missing Combine support in UITableView and I would like to keep this approach here.

**Pull request merged**